configs/components/_base-ruby-selinux.rb

```diff
@@ -57,6 +68,11 @@
     steps << "#{platform.patch} --strip=0 --fuzz=0 --ignore-whitespace --no-backup-if-mismatch < ../selinuxswig_ruby_wrap.patch"
   end
 
+  # libselinux 3.3 is the minimum version we want to build on RHEL 9, but the
```

Comment:
Have we tried using versions 3.4 or 3.5? I'm fine merging this as is, but, if we could just use a later version of selinux, maybe we can avoid this hack?

Reply:
I haven't tried it, my assumption was that there would be compatibility issues if there were a discrepancy between the bindings we built and the version running on the system. Do you think it would be worth a shot?

Reply:
Yeah, I'm not sure if there would be a discrepancy, seems like it might possibly work? As long as the major versions are the same...

Reply:
I tried using the SELinux 3.4 tarball and installing the libselinux-devel-3.4 package. It seems like it creates the same compatibility issue as we saw in the original ticket. On an adhoc agent build on x86_64:
```
Begin tests/selinux.rb

PA-3067: Manage selinux

  * test require 'selinux'

  ergotropic-trap.delivery.puppetlabs.net (ergotropic-trap.delivery.puppetlabs.net) 23:22:11$ env PATH="/opt/puppetlabs/puppet/bin:${PATH}" ruby -e 'require "selinux"'
 /opt/puppetlabs/puppet/lib/ruby/site_ruby/2.7.0/rubygems/core_ext/kernel_require.rb:86:in `require': /lib64/libselinux.so.1: version `LIBSELINUX_3.4' not found (required by /opt/puppetlabs/puppet/lib/ruby/vendor_ruby/2.7.0/x86_64-linux/selinux.so) - /opt/puppetlabs/puppet/lib/ruby/vendor_ruby/2.7.0/x86_64-linux/selinux.so (LoadError)
    	from /opt/puppetlabs/puppet/lib/ruby/site_ruby/2.7.0/rubygems/core_ext/kernel_require.rb:86:in `require'
    	from -e:1:in `<main>'
```
I assume this happens because there's a mismatch between what we build against and what's present on the system.

I think the only way we can avoid this issue is to target the lowest common denominator, SELinux 3.3.